digdag-core/src/main/java/io/digdag/core/agent/InProcessTaskCallbackApi.java

```diff
@@ -96,23 +100,24 @@ public TaskLogger newTaskLogger(TaskRequest request)
     public void taskHeartbeat(int siteId,
             List<String> lockedIds, AgentId agentId, int lockSeconds)
     {
-        queueClient.taskHeartbeat(siteId, lockedIds, agentId.toString(), lockSeconds);
+        tm.begin(() -> queueClient.taskHeartbeat(siteId, lockedIds, agentId.toString(), lockSeconds));
     }
 
     @Override
     public Optional<StorageObject> openArchive(TaskRequest request)
-        throws IOException
     {
         if (!request.getRevision().isPresent()) {
             return Optional.absent();
         }
         String revision = request.getRevision().get();
 
         try {
-            return archiveManager.openArchive(
-                    pm.getProjectStore(request.getSiteId()),
-                    request.getProjectId(),
-                    revision);
+            return tm.begin((TransactionManager.SupplierInTransaction<
+                            Optional<StorageObject>, ResourceNotFoundException,
+                            StorageFileNotFoundException, RuntimeException>) () ->
```

Comment:
This is very minor but Instead of casting the lambda type, you can use this syntax:

```java
tm.<Optional<StorageObject>, ResourceNotFoundException, StorageFileNotFoundException>begin(() -> {
  ...
},
ResourceNotFoundException.class, StorageFileNotFoundException.class);
```

I wish that javac becomes smart enough to guess the types automatically...